hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java

```diff
@@ -219,17 +233,19 @@ public void register(ResultSet resultSet, Statement statement) {
 			}
 		}
 		if ( statement != null ) {
-			HashMap<ResultSet,Object> resultSets = xref.get( statement );
+			final StatementRegistryEntry statementRegistryEntry = xref.remove( statement );
+			HashMap<ResultSet,Object> resultSets = statementRegistryEntry != null ?
+					statementRegistryEntry.resultSets : null;
 
 			// Keep this at DEBUG level, rather than warn.  Numerous connection pool implementations can return a
 			// proxy/wrapper around the JDBC Statement, causing excessive logging here.  See HHH-8210.
-			if ( resultSets == null ) {
+			if ( resultSets == null) {
```

Comment:
Seems the above code change is not better than before. We do need a space char here.

Reply:
That's weird. I thought I ran a CTRL ALT L on those bocks, and the `check` command didn't complain either.

I'll fix all issues and push a new commit.